components/ProgressView/tests/unit/ProgressViewProgressWithCurrentScreenScaleTests.m

```diff
@@ -0,0 +1,119 @@
+
+#import "MaterialMath.h"
+#import "MaterialProgressView.h"
+
+@interface MDCProgressView ()
+@property(nonatomic, strong) UIView *progressView;
+@end
+
+@interface ProgressViewProgressWithCurrentScreenScaleTests : XCTestCase
+@property(nonatomic, strong) UIWindow *window;
+@property(nonatomic, strong) MDCProgressView *progressView;
+@end
+
+@implementation ProgressViewProgressWithCurrentScreenScaleTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.window = [[UIWindow alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
+  self.progressView = [[MDCProgressView alloc] initWithFrame:CGRectMake(0, 0, 100, 10)];
+  [self.window addSubview:self.progressView];
+}
+
+- (void)tearDown {
+  self.progressView = nil;
+
+  [super tearDown];
+}
+
+- (void)testZero {
+  // When
+  self.progressView.progress = (float)0.00;
+  [self.progressView layoutIfNeeded];
+
+  // Then
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.x, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.y, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.size.width, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.size.height, 10, 0.001);
+}
+
+- (void)test50 {
+  // When
+  self.progressView.progress = (float)0.50;
+  [self.progressView layoutIfNeeded];
+
+  // Then
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.x, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.y, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.size.width, 50, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.size.height, 10, 0.001);
+}
+
+- (void)test65 {
+  // When
+  self.progressView.progress = (float)0.65;
+  [self.progressView layoutIfNeeded];
+
+  // Then
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.x, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.y, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.size.width, 65, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.size.height, 10, 0.001);
+}
+
+- (void)test97 {
+  // When
+  self.progressView.progress = (float)0.97;
+  [self.progressView layoutIfNeeded];
+
+  // Then
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.x, 0, 0.001);
+  XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.origin.y, 0, 0.001);
+  CGFloat screenScale = self.progressView.window.screen.scale;
+
+  if (MDCCGFloatEqual(screenScale, 1)) {
+    // Note the surprising value of 98 here; this is due to the use of MDCCeil to calculate
+    // screen-aligned pixel values for the progress view's frame.
+    // TODO(https://github.com/material-components/material-components-ios/issues/9482): This should
+    // be 97.
+    XCTAssertEqualWithAccuracy(self.progressView.progressView.frame.size.width, 98, 0.001);
```

Comment:
This is 97 when I test it on an iPad 2 and iPhone 4s, but 98 on 64-bit devices. Given its fragile nature, I'm not sure if it's worth keeping or not.  What do you think?

Reply:
Also looks like our test runners aren’t covering this case unfortunately.

Reply:
They're not, but this is the type of scenario where I believe we still rely on maintainers/authors to conduct sufficient testing.

Reply:
Looks like we could add 32-bit simulator testing if https://github.com/google/xctestrunner/issues/14 is fixed.

Reply:
> They're not, but this is the type of scenario where I believe we still rely on maintainers/authors to conduct sufficient testing.

+1, but I'm unfortunately not aware of a way to do so from Catalina and our test infra should be giving us comprehensive enough coverage for behavior deltas like this.

All that being said, I'm going to move these unit tests out to a separate PR so we can discuss further there.

Reply:
Catalina is a problem, but right now I believe [our supported Xcode version is Xcode 10](https://github.com/material-components/material-components-ios/tree/develop/docs/build-env#xcode).  I think we should try to identify a way to get 32-bit unit tests running in CI until we drop iOS 9.